controllers/gce/controller/cluster_manager.go

```diff
@@ -159,7 +162,7 @@ func (c *ClusterManager) Checkpoint(lbs []*loadbalancers.L7RuntimeInfo, nodeName
 	for _, p := range fwNodePorts {
 		np = append(np, p.Port)
 	}
-	if err := c.firewallPool.Sync(np, nodeNames); err != nil {
+	if err := c.firewallPool.Sync(np, nodeNames, currentIngress); err != nil {
```

Comment:
I am not sure if it will work but want to throw an idea: How about letting firewallPool.Sync return an error and cluster manager can then decide to generate an event for it.

That way wont need to pass currentIngress and eventrecorder down to FirewallPool. In future, the same logic can be extended by other libraries like healthcheck and forwarding rules to generate events as well? Better than passing extra objects everywhere